src/main/java/com/mixpanel/android/mpmetrics/MixpanelPushNotification.java

```diff
@@ -187,7 +190,7 @@ protected void buildNotificationFromData() {
     protected Notification createNotification(Intent inboundIntent) {
         this.parseIntent(inboundIntent);
 
-        if (this.data == null) {
+        if (this.data == null || this.data.silent) {
```

Comment:
it'd be useful to put this into it's own conditional just so we can log a message that says why we aren't showing a message. E.g. `MPLog.d(LOGTAG, "Notification will not be shown because 'mp_silent = true'")`